### ISSUE TITLE
Error while trying to use expo-payments-stripe 8.0.0

### ISSUE THREAD

**Issue by graemeharnish** [expo/expo#6670]
While using v8 of stripe payments I get ....

```
While trying to resolve module `expo-payments-stripe` from file `x`, the package `x` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`x`. Indeed, none of these files exist:

  * /x(.native|.android.expo.ts|.native.expo.ts|.expo.ts|.android.expo.tsx|.native.expo.tsx|.expo.tsx|.android.expo.js|.native.expo.js|.expo.js|.android.expo.jsx|.native.expo.jsx|.expo.jsx|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx|.android.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json|.android.wasm|.native.wasm|.wasm)
```
Previous versions do not throw that error.

Thanks!

**Comment by graemeharnish** [comment]
Ah, looks like it was missing https://github.com/expo/expo/blob/master/packages/expo-payments-stripe/index.js adding that and fixing the path works

**Comment by kopax** [comment]
Hi @graemeharnish I was expecting to use stripe as well, could you describe the `expo-payments-stripe`  you are referring to as `last working`, `broken` and the fix you have applied?

**Comment by cruzach** [comment]
hi @graemeharnish I just tried installing v8 of `expo-payments-stripe` and the index.js file was included. Is it possible that something went wrong installing? Not sure how to repro this

**Comment by graemeharnish** [comment]
Thanks for the responses.  So after an npm install my folders look like this:

README.md      build          ios            src            unimodule.json
android        index.js       package.json   tsconfig.json
╭─ ~/workspace/BellumApp  ‹master*› 
╰─$ ls node_modules/expo-payments-stripe/build
Stripe.d.ts       Stripe.js.map     errorCodes.js     utils
Stripe.js         errorCodes.d.ts   errorCodes.js.map
╭─~/workspace/BellumApp  ‹master*› 

but it's saying it can't find index.js under /node_modules/expo-payments-stripe/build/index.js

It does exist  but in the parent folder.  Am I doing something wrong here?

Thanks!

**Comment by kopax** [comment]
That look like a version missmatch more than an error in `expo-payment-stripe`, but perhaps.

**Comment by graemeharnish** [comment]
Interesting error messages? :)
```

├─┬ expo-payments-stripe@8.0.0
npm ERR! peer dep missing: react-native-vector-icons@>6.0.0, required by react-native-elements@1.2.7
npm ERR! peer dep missing: react-native-vector-icons@*, required by react-native-paper@3.4.0
npm ERR! extraneous: node-pre-gyp@0.14.0 node_modules/node-pre-gyp
npm ERR! invalid: react-native@0.61.4 node_modules/react-native
npm ERR! extraneous: @react-native-community/cli@3.0.4 node_modules/@react-native-community/cli
npm ERR! extraneous: @react-native-community/cli-platform-android@3.0.3 node_modules/@react-native-community/cli-platform-android
npm ERR! extraneous: @react-native-community/cli-platform-ios@3.0.0 node_modules/@react-native-community/cli-platform-ios
npm ERR! extraneous: abort-controller@3.0.0 node_modules/abort-controller
npm ERR! extraneous: art@0.10.3 node_modules/art
npm ERR! extraneous: base64-js@1.3.1 node_modules/base64-js
npm ERR! extraneous: connect@3.7.0 node_modules/connect
npm ERR! extraneous: event-target-shim@5.0.1 node_modules/event-target-shim
npm ERR! extraneous: fbjs-scripts@1.2.0 node_modules/fbjs-scripts
npm ERR! extraneous: hermes-engine@0.2.1 node_modules/hermes-engine
npm ERR! extraneous: jsc-android@245459.0.0 node_modules/jsc-android
npm ERR! extraneous: metro-babel-register@0.56.3 node_modules/metro-babel-register
npm ERR! extraneous: metro-react-native-babel-transformer@0.56.3 node_modules/metro-react-native-babel-transformer
npm ERR! extraneous: metro-source-map@0.56.3 node_modules/metro-source-map
npm ERR! extraneous: pretty-format@24.9.0 node_modules/pretty-format
npm ERR! extraneous: react-devtools-core@3.6.3 node_modules/react-devtools-core
npm ERR! extraneous: stacktrace-parser@0.1.8 node_modules/stacktrace-parser
```

**Comment by kopax** [comment]
You should try to fix them, start with the first two `ERR`.

**Comment by graemeharnish** [comment]
So I tried creating a brand new project via command line.  Same error
```

╰─$ npm ls | grep stripe
├─┬ expo-payments-stripe@8.0.0
npm ERR! peer dep missing: react-native-screens@^1.0.0 || ^1.0.0-alpha, required by react-navigation-stack@1.10.3
npm ERR! peer dep missing: react-native-screens@^1.0.0 || ^1.0.0-alpha, required by react-navigation-tabs@2.6.2
npm ERR! invalid: react-native@0.61.4 /Users/graemeharnish/workspace/my-new-proj/node_modules/react-native
npm ERR! extraneous: @react-native-community/cli@3.0.4 /Users/graemeharnish/workspace/my-new-proj/node_modules/@react-native-community/cli
npm ERR! extraneous: @react-native-community/cli-platform-android@3.0.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/@react-native-community/cli-platform-android
npm ERR! extraneous: @react-native-community/cli-platform-ios@3.0.0 /Users/graemeharnish/workspace/my-new-proj/node_modules/@react-native-community/cli-platform-ios
npm ERR! extraneous: abort-controller@3.0.0 /Users/graemeharnish/workspace/my-new-proj/node_modules/abort-controller
npm ERR! extraneous: art@0.10.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/art
npm ERR! extraneous: base64-js@1.3.1 /Users/graemeharnish/workspace/my-new-proj/node_modules/base64-js
npm ERR! extraneous: connect@3.7.0 /Users/graemeharnish/workspace/my-new-proj/node_modules/connect
npm ERR! extraneous: event-target-shim@5.0.1 /Users/graemeharnish/workspace/my-new-proj/node_modules/event-target-shim
npm ERR! extraneous: fbjs-scripts@1.2.0 /Users/graemeharnish/workspace/my-new-proj/node_modules/fbjs-scripts
npm ERR! extraneous: hermes-engine@0.2.1 /Users/graemeharnish/workspace/my-new-proj/node_modules/hermes-engine
npm ERR! extraneous: jsc-android@245459.0.0 /Users/graemeharnish/workspace/my-new-proj/node_modules/jsc-android
npm ERR! extraneous: metro-babel-register@0.56.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/metro-babel-register
npm ERR! extraneous: metro-react-native-babel-transformer@0.56.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/metro-react-native-babel-transformer
npm ERR! extraneous: metro-source-map@0.56.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/metro-source-map
npm ERR! extraneous: react-devtools-core@3.6.3 /Users/graemeharnish/workspace/my-new-proj/node_modules/react-devtools-core
npm ERR! extraneous: stacktrace-parser@0.1.8 /Users/graemeharnish/workspace/my-new-proj/node_modules/stacktrace-parser
```
and my package.json
```
╰─$ cat package.json 
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "test": "jest --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/samples": "~36.0.0",
    "@expo/vector-icons": "~10.0.0",
    "@react-navigation/web": "~1.0.0-alpha.9",
    "expo": "~36.0.0",
    "expo-asset": "~8.0.0",
    "expo-constants": "~8.0.0",
    "expo-font": "~8.0.0",
    "expo-payments-stripe": "~8.0.0",
    "expo-web-browser": "~8.0.0",
    "react": "~16.9.0",
    "react-dom": "~16.9.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz",
    "react-native-gesture-handler": "~1.5.0",
    "react-native-reanimated": "~1.4.0",
    "react-native-screens": "2.0.0-alpha.12",
    "react-native-web": "~0.11.7",
    "react-navigation": "~4.0.10",
    "react-navigation-stack": "~1.10.3",
    "react-navigation-tabs": "~2.6.2"
  },
  "devDependencies": {
    "@babel/core": "^7.0.0",
    "babel-preset-expo": "~8.0.0",
    "jest-expo": "~36.0.1"
  },
  "private": true
}
```
Note:  You have to import the library before the error occurs.  Thanks for the help!

**Comment by eclatsujan** [comment]
Is it fixed? I still have same issue. I have temporarily fixed the issue by replacing main:build/index.js to index.js.  There is no index file in build folder but package.json is pointing toward it.

**Comment by cruzach** [comment]
Fix has been released in `expo-payments-stripe` 8.0.1
